hoomd/md/MeshDynamicBondUpdater.cc

```diff
@@ -0,0 +1,411 @@
+
+            Scalar3 dcd;
+            dcd.x = h_pos.data[idx_c].x - h_pos.data[idx_d].x;
+            if (dcd.x > boxL.x || dcd.x < -boxL.x)
+                continue;
+            dcd.y = h_pos.data[idx_c].y - h_pos.data[idx_d].y;
+            if (dcd.y > boxL.y || dcd.y < -boxL.y)
+                continue;
+            dcd.z = h_pos.data[idx_c].z - h_pos.data[idx_d].z;
+            if (dcd.y > boxL.y || dcd.y < -boxL.y)
+                continue;
+            }
+
+        unsigned int tag_a = bond.tag[0];
+        unsigned int tag_b = bond.tag[1];
+
+        unsigned int idx_a = h_rtag.data[tag_a];
+        unsigned int idx_b = h_rtag.data[tag_b];
+
+        unsigned int test_idx = idx_c;
+
+        for (unsigned int ii = 0; ii < h_n_meshbond.data[idx_c]; ii++)
+            {
+            if (h_meshbond_pos_list.data[table_indexer(idx_c, ii)] > 1)
+                continue;
+            test_idx = h_meshbondlist.data[table_indexer(idx_c, ii)].idx[0];
+            if (test_idx == idx_d)
+                break;
+            }
+
+        if (test_idx == idx_d)
+            continue;
+
+        unsigned int tr_idx1 = h_neigh_bonds.data[i].x;
+        unsigned int tr_idx2 = h_neigh_bonds.data[i].y;
+
+        typename Angle::members_t& triangleT = h_triangles.data[tr_idx1];
+
+        for (unsigned int j = 0; j < 3; j++)
+            {
+            if (triangleT.tag[j] == tag_d)
+                {
+                tr_idx1 = h_neigh_bonds.data[i].y;
+                tr_idx2 = h_neigh_bonds.data[i].x;
+                break;
+                }
+            }
+
+        typename Angle::members_t& triangle1 = h_triangles.data[tr_idx1];
+
+        unsigned int iterator = 0;
+
+        bool a_before_b = false;
+
+        while (tag_a != triangle1.tag[iterator])
+            iterator++;
+
+        iterator = (iterator + 1) % 3;
+
+        if (tag_b == triangle1.tag[iterator])
+            {
+            a_before_b = true;
+            iterator = (iterator + 1) % 3;
+            }
+
+        unsigned int type_id = h_typeval.data[i].type;
+
+        Scalar energyDifference = 0;
+
+        unsigned int idx_cc = idx_d;
+        unsigned int idx_dd = idx_c;
+
+        if (a_before_b)
+            {
+            idx_cc = idx_c;
+            idx_dd = idx_d;
+            }
+
+        bool have_to_check_surrounding = false;
+        for (auto& force : m_forces)
+            {
+            energyDifference += force->energyDiff(idx_a, idx_b, idx_cc, idx_dd, type_id);
+            if (force->checkSurrounding())
+                have_to_check_surrounding = true;
+            }
+
+        // Initialize the RNG
+        RandomGenerator rng(hoomd::Seed(RNGIdentifier::MeshDynamicBondUpdater, timestep, seed),
+                            hoomd::Counter(i));
+
+        // compute the random force
+        UniformDistribution<Scalar> uniform(0, Scalar(1));
+
+        Scalar rand_number = uniform(rng);
+        Scalar part_func = exp(-m_inv_T * energyDifference);
```

Comment:
m_inv_kT perhaps?

Readers would expect T to be in Kelvin or some other temperature unit. Naming it kT makes it clear that this is a thermal energy.